src/dialects/mysql/index.js

```diff
@@ -104,7 +104,14 @@ assign(Client_MySQL.prototype, {
     return new Promise((resolver, rejecter) => {
       stream.on('error', rejecter)
       stream.on('end', resolver)
-      connection.query(queryOptions, obj.bindings).stream(options).pipe(stream)
+      const queryStream = connection.query(queryOptions, obj.bindings).stream(options)
+
+      queryStream.on('error', (err) => {
```

Comment:
Is it only relevant for MySQL?

Reply:
I'm not sure, I could reproduce the bug only in MySQL because that's the database I am using.